### PR TITLE
feat(ali-oss): add listV2() function.

### PR DESCRIPTION
**Pull request by SlahserZ**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## ali-oss ([<kbd>unpkg</kbd>](https://unpkg.com/browse/ali-oss@latest/))
was missing the following properties:
1. initOptions
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against b084b916a74c3e09fd8eeda94d20b546fd799068
</p>

File: types/ali-oss/index.d.ts [hunk 1]
```diff
@@ -990,6 +990,11 @@ declare class OSS {
      */
     list(query: OSS.ListObjectsQuery | null, options: OSS.RequestOptions): Promise<OSS.ListObjectResult>;
 
+    /**
+     * List Objects in the bucket.(V2)
+     */
+    listV2(query: OSS.ListV2ObjectsQuery | null, options: OSS.RequestOptions): Promise<OSS.ListObjectResult>;
```
Comment (gabritto) [inline types/ali-oss/index.d.ts:996]: Should `options` be optional? There's a default value in the implementation: https://github.com/ali-sdk/ali-oss/blob/a759699be8c7e4acbdc72a5f8d121f2224a7f846/lib/object.js#L243, and the readme says it's optional.

**Review by gabritto — CHANGES_REQUESTED** [review]

File: types/ali-oss/index.d.ts (hunk 1 above)
Comment (SlahserZ) [reply types/ali-oss/index.d.ts:996]: yes, you are right, thanks.

**Review by RyanCavanaugh — APPROVED** [review]

**Comment by SlahserZ** [comment]
Ready to merge

**Pull request merged**